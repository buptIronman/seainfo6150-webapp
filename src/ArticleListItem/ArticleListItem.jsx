import React,{useState} from 'react'
import HTMLText from "../HTMLText/HTMLText";
import ArticleTextToggleButton from "../ArticleTextToggleButton/ArticleTextToggleButton";
import styles from './ArticleListItem.module.css';

const ArticleListItem = (props) => {
    const [status, setStatus] = useState(false);

    const onClick = (buttonState) => {
        console.log(buttonState+ "item")
        setStatus(!buttonState);
    }
    

    if(status){
        return (<div className={styles.item}>
            <h1 className={styles.title}>{props.article.title}</h1>
            <HTMLText className={styles.text} text={props.article.shortText} >{props.article.shortText}</HTMLText>
            <br/>
            <time dateTime={props.article.displayDate} className={styles.time}>{props.article.displayDate}</time>
            <ArticleTextToggleButton onClick={onClick} ButtonText="show less"></ArticleTextToggleButton>
        </div>)}
    else
        return(<div className={styles.item}>
            <h1 className={styles.title}>{props.article.title}</h1>
            <ArticleTextToggleButton  onClick={onClick} ButtonText="show more"></ArticleTextToggleButton>
        </div>)


}

export default ArticleListItem