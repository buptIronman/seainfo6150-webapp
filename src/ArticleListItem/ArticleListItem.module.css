.item{
    margin: 30px;
    border-style: solid;
    border-color: red;
    border-radius: 15px;
    background-color: red;
    background-image: linear-gradient(to right, brown, white 20%);
    font-size: medium;
    padding: 10px;
    display: -ms-grid;
}

.title{
    color: firebrick;
    text-decoration: underline;
}

.time{
    font-size: small;
    font-weight: bolder;
}

.text{
    font-size: x-small;
    padding-bottom: 10px;
}

@media screen and (min-width: 900px) {
    article {
        padding: 1rem 3rem;
    }
}