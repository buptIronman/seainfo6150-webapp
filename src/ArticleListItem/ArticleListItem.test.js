import React from 'react';
import ArticleListItem from "./ArticleListItem";
import AddingMachineHooks from "../component-examples/AddingMachine/AddingMachineHooks";

describe("ArticleListItem test", () => {
    it("no data", () => {
        const { container } = render(<ArticleListItem article={0} />);
        expect(container).toMatchSnapshot();
    });

    it("renders correctly when currentNum is 10", () => {
        const { container } = render(<AddingMachineHooks startNum={10} />);
        expect(container).toMatchSnapshot();
    });

    it("renders correctly after button is clicked", () => {
        const { container, getByText } = render(
            <AddingMachineHooks startNum={0} />
        );
        const button = getByText("Click me to add 1");
        fireEvent.click(button);
        expect(container).toMatchSnapshot();
    });
})