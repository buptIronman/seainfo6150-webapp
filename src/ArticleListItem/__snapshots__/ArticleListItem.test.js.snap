// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem test no data 1`] = `
<div>
  <div
    class="item"
  >
    <h1
      class="title"
    />
    <div>
      <button
        class="button"
      >
        show more
      </button>
    </div>
  </div>
</div>
`;

exports[`ArticleListItem test renders correctly after button is clicked 1`] = `
<div>
  <div
    style="border: 1px solid black; margin: 2rem;"
  >
    <header />
    <div>
      We started with 
      0
    </div>
    <div>
      We're now at 
      1
    </div>
    <button>
      Click me to add 1
    </button>
  </div>
</div>
`;

exports[`ArticleListItem test renders correctly when currentNum is 10 1`] = `
<div>
  <div
    style="border: 1px solid black; margin: 2rem;"
  >
    <header />
    <div>
      We started with 
      10
    </div>
    <div>
      We're now at 
      10
    </div>
    <button>
      Click me to add 1
    </button>
    <div>
      Whee!
    </div>
  </div>
</div>
`;
